V3-Upgrade-Guide.md

```diff
@@ -36,58 +32,40 @@ Depending on if you're using `npm` or `yarn`, run `npm install` or `yarn` in the
+## Update to Node 18 or Newer
 
-### Adjust Operation Names in OData Generated Clients
+All SAP Cloud SDK for JavaScript libraries now support node 18 (LTS) as the **minimum** node version.
+If you are using a node version older than 18, update your runtime environment to a newer version.
+Note, that the transpilation target of the SDK changed from `es2019` to `es2021`.
 
-Rules for naming of OData operations (actions or functions) in the generated client have been changed.
-This applies to bound and unbound operations.
-If an operation begins with an `underscore` symbol, the `_` will be removed from the resulting generated client code.
-To adjust the names, search in `function-import.ts` and `action-import.ts` files in your generated client code for any operation starting with `_`.
-Similarly, to adjust the names of bound operations of an entity, search in the respective entity's `.ts` file, e.g., `TestEntity.ts`.
+## Replace Removed Functionality
 
-### Check for removed deprecated functions and replace them if required
-
-While the SAP Cloud SDK maintains backwards compatibility within a major version where possible, a new major release breaks compatibility where required to simplify the programming interface.
-Most of the removed functions had been deprecated before, so ideally they are not used anymore.
+Most of the removed functionality had been deprecated in version 2, so ideally they are not used anymore.
 The following sub-sections describe affected modules, functions and interfaces with instructions on how to replace them.
 
-#### Package `@sap-cloud-sdk/http-client`
-
-The overload, that accepted `HttpRequestConfigWithOrigin` as a parameter, is removed and replaced by the function `executeHttpRequestWithOrigin`.
-
-#### Package `@sap-cloud-sdk/util`
+### Package `@sap-cloud-sdk/http-client` <!-- omit from toc -->
 
-The field `logger` on the interface `LoggerOptions` was not used and is removed from the interface.
+- The `executeHttpRequest()` function overload, that accepted `HttpRequestConfigWithOrigin` as a parameter, is removed. Use `executeHttpRequestWithOrigin()` instead.
 
-The function `variadicArgumentToArray` is replaced by the function `transformVariadicArgumentToArray`.
+### Package `@sap-cloud-sdk/util` <!-- omit from toc -->
 
-#### Package `@sap-cloud-sdk/connectivity`
+- The field `logger` on the interface `LoggerOptions` was not used and is removed from the interface. There is no replacement.
+- The function `variadicArgumentToArray` is replaced by the function `transformVariadicArgumentToArray`.
 
-The generic types of `JwtKeyMapping` is simplified so the second type argument `JwtKeysT` are always strings.
+### Package `@sap-cloud-sdk/connectivity` <!-- omit from toc -->
 
-#### Package `@sap-cloud-sdk/odata-common`
+- The generic type of `JwtKeyMapping` was improved, so that the second type argument `JwtKeysT` extends string.
```

Comment:
nit: is this really 'improved'? I chose 'simplified' because we narrowed the type which imo is simpler, not sure if it's "better".

Reply:
Haha, I changed it because narrowing in my opinion makes it more correct, but more restricted, thus less simple. How about:

```suggestion
- The generic type parameter `JwtKeysT` in `JwtKeyMapping` is now narrowed to extend `string`.
```

Alternative, this does not seem like something that is required in an upgrade guide. Maybe we should leave this out.

Reply:
Don't know, I've put it here because it is a breaking change. I'm okay with your suggestion, I guess 'simple' can have different meaning in that case (which is, ironically, not that simple to communicate) :)